android-components/components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt

```diff
@@ -294,7 +295,12 @@ class ReaderViewFeature(
 
     private fun WebExtensionController.createReaderUrl(url: String, id: String): String? {
         val colorScheme = config.colorScheme.name.lowercase(Locale.ROOT)
-        return readerBaseUrl?.let { it + "readerview.html?url=$url&id=$id&colorScheme=$colorScheme" }
+        // Encode the original page url, otherwise when the readerview page will try to
+        // parse the url and retrieve the readerview url params (ir and colorScheme)
+        // the parser may get confused because the original webpage url being interpolated
+        // may also include its own search params non-escaped (See Bug 1860490).
+        val encodedUrl = URLEncoder.encode(url, "UTF-8");
+        return readerBaseUrl?.let { it + "readerview.html?url=$encodedUrl&id=$id&colorScheme=$colorScheme" }
```

Comment:
@boek I added this additional change after having investigated [Bug 1860490](https://bugzilla.mozilla.org/show_bug.cgi?id=1860490) deeply enough to finally see how we were hitting that readerview mode issue.

I've currently rolled it into this pull request to allow @dholbert to be able to also give it a try and confirm this additional change along with the original changes part of this pull request are going to fix [Bug 1860490](https://bugzilla.mozilla.org/show_bug.cgi?id=1860490).

I verified on my side that with the changes from this pull request applied [Bug 1860490](https://bugzilla.mozilla.org/show_bug.cgi?id=1860490) was not hit anymore on my Fenix instance, but I had to go through a VPN to be able to reach the content from the url in [Bug 1860490 comment 0](https://bugzilla.mozilla.org/show_bug.cgi?id=1860490#c0) and so I'd like to confirm that the apk built for this PR does fix it also for @dholbert 

(btw, a huge thank you also goes to @Amejia481 :heart:  for giving me access to a VPN with a US exit note I could use to reproduce it and investigate it).

Having said that, it may be as well reasonable to split it out into its own separate patch and pull request built on top of this one, if we prefer that then I'll be more than happy to do that, let me know wdyt.